checks/frozen_deps.go

```diff
@@ -35,19 +36,115 @@ func FrozenDeps(c *checker.CheckRequest) checker.CheckResult {
 	return checker.MultiCheckAnd(
 		isPackageManagerLockFilePresent,
 		isGitHubActionsWorkflowPinned,
+		isDockerfilePinned,
```

Comment:
Should we consider a using an existing Dockerfile parser like this one - github.com/moby/buildkit/frontend/dockerfile/parser?

Reply:
+1 We should be using docker parser!

Reply:
probably a good idea. We still need to use the regex though, so the docker parsing lib will only give us line arguments. Probably will give us peace of mind :-)
I'll make the changes